### PR TITLE
Spring cleaning and update rivets to latest release

### PR DESCRIPTION
**Pull request by mmlb**
Do some spring cleaning like I've been doing to other repos here.

Also needed to update rivets because we've been pinned to an *unreleased* commit for a long time now. I blame go for not making this obvious. I decided to fix this now so that we can back on track and only use merged code paths. Getting the rivets PR merged and then updating this line comes next.

### CODE REVIEW

**Comment by mmlb** [comment]
This came up when I tried using alloy in a go workspace. I thought workspaces were only for working on shared code bases without having to change deps in go.mod for dependencies specified in go.work but that doesn't seem to be right. It looks all the projects share a workspace (not unlike the old GOPATH days?) because alloy was failing to build when I added it to the go.work file, first because it referred to serverservice not fleetdb and then later the missing const.

**Review by DoctorVin — APPROVED** [review]
I'm not sure why the linter is complaining here.

**Comment by mmlb** [comment]
> I'm not sure why the linter is complaining here.

Welp, I thought that maybe updating go and other stuff would fix this but nope... I'm also unsure whats going on since it works locally for me :cry:

**Comment by DoctorVin** [comment]
> > I'm not sure why the linter is complaining here.
> 
> Welp, I thought that maybe updating go and other stuff would fix this but nope... I'm also unsure whats going on since it works locally for me 😢
> 
> --- edit ---
> 
> Internets thinks we need `go mod tidy` so I've tried that too and looks like that works!

`go mod tidy` is my go-to incantation when I get weird complaints about go.mod/go.sum. Glad that worked.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,35 +13,33 @@ jobs:
...
         with:
           fetch-depth: 0
-      -
-        name: Set up Go
-        uses: actions/setup-go@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
         with:
           go-version: "1.20"
```
Comment (DoctorVin) [inline .github/workflows/release.yml:31]: shouldn't we rely on `go.mod` here?

**Review by DoctorVin — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (mmlb) [reply .github/workflows/release.yml]: ~80 moments later...

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**